openedx/features/edly/middleware.py

```diff
@@ -24,3 +26,31 @@ def process_request(self, request):
         if user_is_authenticated and not user_is_superuser and not user_has_edly_organization_access(request):
             logger.exception('Edly user %s has no access for site %s.' % (request.user.email, request.site))
             raise Http404()
+
+
+class SettingsOverrideMiddleware(object):
+    """
+    Django middleware to override django settings from site configuration.
+    """
+
+    def process_request(self, request):
+        """
+        Override django settings from django site configuration.
+        """
+        current_site = request.site
+        try:
+            current_site_configuration = current_site.configuration
+        except SiteConfiguration.DoesNotExist:
+            logger.warning('Site (%s) has no related site configuration.', current_site)
+            return None
+
+        if current_site_configuration.values:
```

Comment:
Just check `if current_site_configuration:`

Reply:
It is possible that a site configuration is present but there are no values in it. This is for that case.

Reply:
Wouldn't that be handled by `current_site_configuration.get_value('DJANGO_SETTINGS_OVERRIDE', None)` below?

Reply:
no. `get_value` tries to get value from `site_configuration.values` which is `None` in this case so it raises error that can't get from NoneType

Reply:
Ok